tests/trainer/test_sharded_checkpoint.py

```diff
@@ -81,8 +86,11 @@ def get_trainer(
 
 def _compare_optims_between_state_dicts(state_dict1, state_dict2):
     # Check that optim params are equal between checkpoint and in memory optimizer
-    state_dict1_optim_params = state_dict1['optimizers']['Adam']['state']
-    state_dict2_optim_params = state_dict2['optimizers']['Adam']['state']
+    optim_key1 = list(state_dict1['optimizers'].keys()).pop()
```

Comment:
maybe assert these lists are of length 1, just to make the assumption more clear

Reply:
haha we've lost years of our lives checking if the number of optimizers is 1